### PR TITLE
Improve summary output when running `rake test_all`

### PR DESCRIPTION
**Pull request by ohayon**
Before:
screen shot 2016-03-09 at 2 14 00 pm

After:
screen shot 2016-03-09 at 2 10 40 pm

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -102,30 +106,75 @@ task :test_all do
...
+      file_path = original_file_path.sub(".", gem_name)
+      "#{file_path}:#{f["line_number"]}".red + " # #{f["full_description"]}".cyan
+    end
+  end
+
+  puts ("*" * 80).yellow
+  box "#{exceptions.count} repo(s) with test failures: " + repos_with_exceptions.join(", ") unless failure_messages.empty?
+  puts format_failures(failures)
+  puts "\nSummary:\n"
+  puts "Finished in #{duration.round(3)} seconds"
```
Comment (KrauseFx) [inline Rakefile]: That's a lot of code, maybe we should move it into a separate file?

Comment (KrauseFx) [inline Rakefile]: Like the whole block

File: Rakefile [hunk 2]
```diff
@@ -102,30 +106,75 @@ task :test_all do
...
+  example_count = 0
+  duration = 0.0
+
+  GEMS.each do |gem_name|
+    failures[gem_name] = []
+    log = File.join(gem_name, log_file)
+    file_text = File.read(log)
+    log_json = JSON.parse(file_text)
+    results = log_json["examples"]
+    failures[gem_name] += results.select { |r| r["status"] == "failed" }
```
Comment (KrauseFx) [inline Rakefile]: Shouldn't this be `!= "success"`? maybe there is something like `timeout`? Or is it really only those 2 values?

**Comment by KrauseFx** [comment]
This looks great, I added some comments :+1:

File: Rakefile (hunk 2 above)
Comment (mfurtak) [inline Rakefile]: Great point!

**Comment by ohwutup** [comment]
Updated!

screen shot 2016-03-09 at 4 35 16 pm

**Comment by orta** [comment]
I love this output :+1: 

( you could get danger to list the fails too )

**Comment by KrauseFx** [comment]
Working on the danger part :+1:

**Comment by KrauseFx** [comment]
This looks great, however I didn't test it :+1:

**Comment by mfurtak** [comment]
screen shot 2016-03-09 at 9 32 00 pm

**Comment by KrauseFx** [comment]
:rocket: wuhuu

**Pull request merged**